src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -1324,15 +1316,42 @@ .. chosenResolvedItem.Suppressions
                         continue;
                     }
 
+                    // See if a dependency with the same version and suppressions has already been chosen
+                    if (resolvedDependencyGraphItems.TryGetValue(childLibraryDependencyIndex, out ResolvedDependencyGraphItem? childResolvedDependencyGraphItem)
+                        && childResolvedDependencyGraphItem.LibraryRangeIndex == childLibraryRangeIndex
+                        && childResolvedDependencyGraphItem.Suppressions.Count == 1
+                        && childResolvedDependencyGraphItem.Suppressions[0].Count == 0)
+                    {
+                        if (childResolvedDependencyGraphItem.IsCentrallyPinnedTransitivePackage)
+                        {
+                            childResolvedDependencyGraphItem.Parents ??= new HashSet<LibraryRangeIndex>();
+
+                            if (!childResolvedDependencyGraphItem.IsRootPackageReference)
+                            {
+                                // Keep track of the parents of this item
+                                childResolvedDependencyGraphItem.Parents?.Add(currentDependencyGraphItem.LibraryRangeIndex);
+                            }
+                        }
+
+                        continue;
```

Comment:
Wouldn't skipping this package prevent us from doing the downgrade checks for it?

Reply:
Downgrade detection happens during the flattening, where each node is re-visited once. This only affects the resolved graph.